### ISSUE TITLE
Message queue is empty if mobile app closed

### ISSUE THREAD

**Issue by sera527** [berezhinskiy/ecoflow_exporter#74]
Доброго дня! маю River 2 Pro. Користуюся мобільним застосунком на iPhone. Спробував скористатися вашим проектом, але виявив, що дані в графані зʼявляються лише тоді, коли відкриваю застосунок на телефоні. В інший час докер контейнер пише таке:
```
2024-08-28 15:16:01 2024-08-28 12:16:01,800 INFO    Reconnection successful, continuing
2024-08-28 15:16:01 2024-08-28 12:16:01,843 ERROR   Unexpected MQTT disconnection: Unspecified error. Will auto-reconnect
2024-08-28 15:16:07 2024-08-28 12:16:07,962 INFO    Subscribed to MQTT topic /app/device/property/R621ZEB4XF270414
2024-08-28 15:16:10 2024-08-28 12:16:10,510 INFO    Message queue is empty. Assuming that the device is offline
2024-08-28 15:16:20 2024-08-28 12:16:20,514 INFO    Message queue is empty. Assuming that the device is offline
2024-08-28 15:16:30 2024-08-28 12:16:30,516 INFO    Message queue is empty. Assuming that the device is offline
2024-08-28 15:16:40 2024-08-28 12:16:40,519 INFO    Message queue is empty. Assuming that the device is offline
2024-08-28 15:16:50 2024-08-28 12:16:50,522 INFO    Message queue is empty. Assuming that the device is offline
2024-08-28 15:17:00 2024-08-28 12:17:00,525 INFO    Message queue is empty. Assuming that the device is offline
2024-08-28 15:17:10 2024-08-28 12:17:10,528 INFO    Message queue is empty. Assuming that the device is offline
2024-08-28 15:17:11 2024-08-28 12:17:11,810 ERROR   No messages received for 60 seconds. Reconnecting to MQTT
2024-08-28 15:17:11 2024-08-28 12:17:11,831 INFO    Connecting to MQTT Broker mqtt-e.ecoflow.com:8883 using client id ANDROID_609351C6-377F-4A61-BB7B-C5000EC5A0F1_1705207767990456322
2024-08-28 15:17:11 2024-08-28 12:17:11,905 INFO    Reconnection successful, continuing
2024-08-28 15:17:11 2024-08-28 12:17:11,940 ERROR   Unexpected MQTT disconnection: Unspecified error. Will auto-reconnect
2024-08-28 15:17:18 2024-08-28 12:17:18,058 INFO    Subscribed to MQTT topic /app/device/property/R621ZEB4XF270414
2024-08-28 15:17:20 2024-08-28 12:17:20,531 INFO    Message queue is empty. Assuming that the device is offline
2024-08-28 15:17:30 2024-08-28 12:17:30,534 INFO    Message queue is empty. Assuming that the device is offline
2024-08-28 15:17:40 2024-08-28 12:17:40,537 INFO    Message queue is empty. Assuming that the device is offline
```
Буду вдячний, якщо допоможете побороти цю проблему.

**Comment by roman-sakh** [comment]
Це не проблема цього застосунку. Це екофлоу не публікує дані

**Comment by sera527** [comment]
@roman-sakh я так розумію, що застосунок на моєму телефоні змушує мій EcoFlow публікувати дані. Чи можна якось змушувати EcoFlow публікувати дані, як це робить мій телефон?

**Comment by oleua** [comment]
Вітання. Після офіційного оновлення Android до версії 16 з вересневими патчами стикнувся з тією же проблемою: коли застосунок активний, то я отримую дані, як тільки він переходить у сплячий режим - зникають... Що можна зробити?

**Comment by oleua** [comment]
The same at my side, after last updates I am getting no data when the app is idle. That is both about rest and mqtt modes. What can be done to improve it?

UPD: After the last update of the [go-ecoflow-exporter](https://github.com/tess1o/go-ecoflow-exporter/issues/21) the stats works well again on my River 2 Plus.

**Comment by VasylSamoilov** [comment]
Probably this is the root cause: 
https://github.com/tolwi/hassio-ecoflow-cloud/blob/afd2016b39138eebabc4ec319d956c422513759c/custom_components/ecoflow_cloud/api/public_api.py#L15

**Comment by Steelflax** [comment]
Маю таку саму проблему на android + ecoflow delta 2 max

**Comment by PretorianX** [comment]
https://github.com/tess1o/go-ecoflow-exporter/commit/746581fc9a0666dc5b4767a9fdd8694e8b9d81d4 отут додали пінг для того аби черга не вмирала

**Comment by PretorianX** [comment]
> Probably this is the root cause: https://github.com/tolwi/hassio-ecoflow-cloud/blob/afd2016b39138eebabc4ec319d956c422513759c/custom_components/ecoflow_cloud/api/public_api.py#L15

I have tested generation of static client ID, it does not fix the issue

**Comment by alex-larin** [comment]
> [tess1o/go-ecoflow-exporter@746581f](https://github.com/tess1o/go-ecoflow-exporter/commit/746581fc9a0666dc5b4767a9fdd8694e8b9d81d4) отут додали пінг для того аби черга не вмирала

схоже що це навіть не пінг, а просто початковий запит `m.requestLatestQuotas(sn)`. В мене своя імплементація схожого застосунку, але проблема була та сама, і додавання явних пінгів чи KeepAlive не допомогли. А ось після додавання запиту на квоти данні пішли, і вже їдуть 10 хвилин.

**Comment by VasylSamoilov** [comment]
так, працює. rest exporter бачить тільки дельту про, на інші девайси каже Cannot get device quota SN=XXX error="can't get parameters, error code 1006". mqtt бачить всі девайси.

**Comment by alex-larin** [comment]
Доречі, запиляв собі інтеграцію по офіціной докє, і воно просто працює - відразу повертає данні, без запуска аппкі та танців с бубном (принаймні для дельти про).  Я не дуже ретельно перевіряв, але схоже що і рест і мкутт повертають усі ті самі данні.

**Comment by PretorianX** [comment]
> > [tess1o/go-ecoflow-exporter@746581f](https://github.com/tess1o/go-ecoflow-exporter/commit/746581fc9a0666dc5b4767a9fdd8694e8b9d81d4) отут додали пінг для того аби черга не вмирала
> 
> схоже що це навіть не пінг, а просто початковий запит `m.requestLatestQuotas(sn)`. В мене своя імплементація схожого застосунку, але проблема була та сама, і додавання явних пінгів чи KeepAlive не допомогли. А ось після додавання запиту на квоти данні пішли, і вже їдуть 10 хвилин.

Не впевнений шо це саме фікс, адже у мене вже другу добу експортер без змін тупо працює без запущеної апки на телефоні
і після блекауту знов дісконект.

**Comment by oleua** [comment]
> > > [tess1o/go-ecoflow-exporter@746581f](https://github.com/tess1o/go-ecoflow-exporter/commit/746581fc9a0666dc5b4767a9fdd8694e8b9d81d4) отут додали пінг для того аби черга не вмирала
> > 
> > 
> > схоже що це навіть не пінг, а просто початковий запит `m.requestLatestQuotas(sn)`. В мене своя імплементація схожого застосунку, але проблема була та сама, і додавання явних пінгів чи KeepAlive не допомогли. А ось після додавання запиту на квоти данні пішли, і вже їдуть 10 хвилин.
> 
> Не впевнений шо це саме фікс, адже у мене вже другу добу експортер без змін тупо працює без запущеної апки на телефоні і після блекауту знов дісконект.

Те саме бачу... Здається, логіка така, що достатньо один раз "пхнути" з увімкненою апкою і далі більше не злітає конфігурація.
UPD: як наврочив :))) Зайшов на дешборд і помітив, що все по нулях і в режимі Standby, перезавантаження контейнерів помогло. Логіка не зрозуміла.

**Comment by ces3001** [comment]
Same issue with EcoFlow Pro. I can only get live data from while iOS app is running and connected to EcoFlow.  Client_ID is the same always.

**Comment by serhiioliinyk** [comment]
hi everyone. My fresh PR https://github.com/berezhinskiy/ecoflow_exporter/pull/82 contains fix for this issue. Now exporter sends every 15 sec request to `endpoint /app/{user_id}/{device_sn}/thing/property/get`. This keeps
  metrics flowing without requiring the official app to be open

**Comment by denisgolius** [comment]
> hi everyone. My fresh PR [#82](https://github.com/berezhinskiy/ecoflow_exporter/pull/82) contains fix for this issue. Now exporter sends every 15 sec request to `endpoint /app/{user_id}/{device_sn}/thing/property/get`. This keeps metrics flowing without requiring the official app to be open

I'd like to have a note for PRs made with Cloude or other AI tooling.

**Comment by denisgolius** [comment]
> hi everyone. My fresh PR [#82](https://github.com/berezhinskiy/ecoflow_exporter/pull/82) contains fix for this issue. Now exporter sends every 15 sec request to `endpoint /app/{user_id}/{device_sn}/thing/property/get`. This keeps metrics flowing without requiring the official app to be open

Considering that you haven't even written a single line of code or made a single commit to the repository, how can you be sure that Cloude has actually solved the problem?
